main.go

```diff
@@ -0,0 +1,203 @@
+
+	logging.Info(logger).Log(logging.MessageKey(), fmt.Sprintf("%s is up and running!", applicationName), "elapsedTime", time.Since(start))
+
+	// The actual logic
+	_, runnable, done := codex.Prepare(logger, nil, metricsRegistry, rootRouter)
+
+	waitGroup, shutdown, err := concurrent.Execute(runnable)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to start device manager: %s\n", err)
+		return 1
+	}
+	webhookFactory.SetExternalUpdate(createArgusSynchronizer(svc, logger))
+
+	// wait for DNS to propagate before subscribing to SNS
+	if err = webhookFactory.DnsReady(); err == nil {
+		logging.Info(logger).Log(logging.MessageKey(), "server is ready to take on subscription confirmations")
+		webhookFactory.PrepareAndStart()
+	} else {
+		logging.Error(logger).Log(logging.MessageKey(), "Server was not ready within a time constraint. SNS confirmation could not happen",
+			logging.ErrorKey(), err)
+	}
+
+	signals := make(chan os.Signal, 10)
+	signal.Notify(signals)
+	for exit := false; !exit; {
+		select {
+		case s := <-signals:
+			if s != os.Kill && s != os.Interrupt {
+			} else {
+				logging.Error(logger).Log(logging.MessageKey(), "exiting due to signal", "signal", s)
+				exit = true
+			}
+		case <-done:
+			exit = true
+		}
+	}
+	close(shutdown)
+	close(webhookRegistry.Changes)
+	waitGroup.Wait()
+
+	return 0
+}
+
+func createArgusSynchronizer(svc xwebhook.Service, logger log.Logger) func([]webhook.W) {
+	return func(webhooks []webhook.W) {
+		for _, w := range webhooks {
+			logging.Info(logger).Log("msg", "Pushing webhook update from SNS into Argus")
+
+			webhook, err := toNewWebhook(&w)
+			if err != nil {
+				logging.Error(logger).Log(logging.MessageKey(), "could not convert to new webhook struct", "err", err)
+				continue
+			}
+			err = svc.Add("Argus", webhook)
+			if err != nil {
+				logging.Error(logger).Log(logging.MessageKey(), "could not add webhook to argus", "err", err)
+			}
+
+		}
+	}
+}
+
+func printVersion(f *pflag.FlagSet, arguments []string) (error, bool) {
+	printVer := f.BoolP("version", "v", false, "displays the version number")
+	if err := f.Parse(arguments); err != nil {
+		return err, true
+	}
+
+	if *printVer {
+		printVersionInfo(os.Stdout)
+		return nil, true
+	}
+	return nil, false
+}
+
+func printVersionInfo(writer io.Writer) {
+	fmt.Fprintf(writer, "%s:\n", applicationName)
+	fmt.Fprintf(writer, "  version: \t%s\n", Version)
+	fmt.Fprintf(writer, "  go version: \t%s\n", runtime.Version())
+	fmt.Fprintf(writer, "  built time: \t%s\n", BuildTime)
+	fmt.Fprintf(writer, "  git commit: \t%s\n", GitCommit)
+	fmt.Fprintf(writer, "  os/arch: \t%s/%s\n", runtime.GOOS, runtime.GOARCH)
+}
+
+func exitIfError(logger log.Logger, err error) {
+	if err != nil {
+		if logger != nil {
+			logging.Error(logger, emperror.Context(err)...).Log(logging.ErrorKey(), err.Error())
+		}
+		fmt.Fprintf(os.Stderr, "Error: %#v\n", err.Error())
+		os.Exit(1)
+	}
+}
+
+func toNewWebhook(w *webhook.W) (*xwebhook.Webhook, error) {
```

Comment:
Just curious - why are you converting webhooks to a different kind of webhook here?

Reply:
The structs are identical as JSON but not as Go structs

Reply:
Also, I wanted to drop any dependencies on  the `webhook` lib from `xwebhook`